File: library/src/main/java/io/fotoapparat/parameter/Flash.java
package io.fotoapparat.parameter;

/**
 * Flash modes which camera can use.
 */
public enum Flash {

    /**
     * Camera flash will not fire.
     */
    OFF,

    /**
     * Camera flash will always fire regardless light conditions.
     */
    ON,

    /**
     * Camera flash will fire only in low light conditions.
     */
    AUTO,

    /**
     * If deemed necessary by the camera device, a red eye reduction flash will fire during the
     * precapture sequence in low light conditions.
     */
    AUTO_RED_EYE,

    /**
     * Transition flash to continuously on.
     */
    TORCH

}

File: fotoapparat/src/main/java/io/fotoapparat/parameter/Flash.kt
package io.fotoapparat.parameter

/**
 * Flash modes which camera can use.
 */
sealed class Flash : Parameter {

    /**
     * Camera flash will not fire.
     */
    object Off : Flash() {
        override fun toString(): String = "Flash.Off"
    }

    /**
     * Camera flash will always fire regardless light conditions.
     */
    object On : Flash() {
        override fun toString(): String = "Flash.On"
    }

    /**
     * Camera flash will fire only in low light conditions.
     */
    object Auto : Flash() {
        override fun toString(): String = "Flash.Auto"
    }

    /**
     * If deemed necessary by the camera device, a red eye reduction flash will fire during the
     * precapture sequence in low light conditions.
     */
    object AutoRedEye : Flash() {
        override fun toString(): String = "Flash.AutoRedEye"
    }

    /**
     * Transition flash to continuously on.
     */
    object Torch : Flash() {
        override fun toString(): String = "Flash.Torch"
    }

}